src/class/eventManagement/dispatcher/events.class.ts

```diff
@@ -0,0 +1,266 @@
+    eventsValidationFn?: eventsValidationFn<T>;
+    customValidationCbFn?: customValidationCbFn<T>;
+  };
+  standardLog?: StandardLog<T>;
+  parentLogger: Partial<Logger> & Pick<Logger, "info" | "warn" | "error">;
+}
+
+export class EventsHandler<T extends GenericEvent> extends EventEmitter {
+  readonly privateUUID: string;
+  readonly dispatcherChannelName: string;
+
+  #eventsValidationFn: eventsValidationFn<T>;
+  #customValidationCbFn: customValidationCbFn<T>;
+
+  #logger: Partial<Logger> & Pick<Logger, "info" | "warn" | "error">;
+  #standardLogFn: StandardLog<T>;
+
+  constructor(options: EventsHandlerOptions<T>) {
+    super();
+
+    Object.assign(this, { ...options });
+
+    this.#logger = options.parentLogger.child({ module: "events-handler" });
+    this.#standardLogFn = options.standardLog ?? defaultStandardLog;
+
+    this.#eventsValidationFn = options.eventsValidation?.eventsValidationFn ?? new Map();
+    this.#customValidationCbFn = options.eventsValidation?.customValidationCbFn;
+
+    for (const [eventName, validationSchema] of Object.entries(eventsSchema)) {
+      this.#eventsValidationFn.set(eventName, ajv.compile(validationSchema));
+    }
+  }
+
+  public async dispatch(options: DispatchEventOptions<T>): Promise<void> {
+    const { channel, store, redisMetadata, event } = options;
+
+    const transaction = await store.setTransaction({
+      ...event,
+      redisMetadata: {
+        ...event.redisMetadata,
+        ...redisMetadata
+      } as any
+    });
+
+    await channel.publish({
+      ...event,
+      redisMetadata: {
+        ...event.redisMetadata,
+        eventTransactionId: redisMetadata.eventTransactionId,
+        transactionId: transaction.redisMetadata.transactionId
+      }
+    });
+  }
+
+  public async handleEvents(
+    channel: string,
+    event: AnyDispatcherChannelMessage |
+    AnyIncomerChannelMessage<T>
+  ) {
+    if (!event.name || !event.redisMetadata) {
+      throw new Error("Malformed message");
+    }
+
+    if (event.redisMetadata.origin === this.privateUUID) {
+      return;
+    }
+
+    if (channel === this.dispatcherChannelName) {
+      if (!isDispatcherChannelMessage(event)) {
+        throw new Error("Unknown event on Dispatcher Channel");
+      }
+
+      try {
+        this.dispatcherChannelMessagesSchemaValidation(event);
+      }
+      catch (error) {
+        this.#logger.error(this.#standardLogFn({
+          ...event,
+          redisMetadata: {
+            ...event.redisMetadata,
+            origin: this.privateUUID
+          }
+        } as unknown as StandardLogOpts<T>)(error.stack));
+      }
+
+      match<DispatcherChannelEvents>({ name: event.name })
+        .with({ name: "REGISTER" }, async() => {
+          this.#logger.info(this.#standardLogFn(event as any)("Registration asked"));
+
+          this.emit("APPROVEMENT", event);
+        })
+        .otherwise(() => {
+          throw new Error("Unknown event on Dispatcher Channel");
+        });
```

Comment:
Est-ce que ça marche?

Reply:
it wasn't binded to the context like with exhaustive so I added a catch a top level